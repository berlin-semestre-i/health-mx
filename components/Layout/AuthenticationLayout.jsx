import React from 'react'
import { Grid, Container } from 'semantic-ui-react'
import styled from 'styled-components'
import media from 'styled-media-query'

const AuthenticationLayout = ({ children }) => (
  <AuthenticationGrid>
    <ImageColumn className="image-container" width={9}/>
    <FormColumn mobile={16} tablet={7} computer={7}>
      <FormContainer>
        { children }
      </FormContainer>
    </FormColumn>
  </AuthenticationGrid>
)

export default AuthenticationLayout

const AuthenticationGrid = styled(Grid)`
  && {
    height: 100vh;
    margin: 0;
    margin-left: -84px;
  }
`
const ImageColumn = styled(Grid.Column)`
  ${media.greaterThan('large')`
    &&&.image-container {
      padding: 0;
      background-image: url("../../static/images/enfermera.png");
      background-repeat: no-repeat;
      background-size: 100%;
    }
  `}

  ${media.lessThan('large')`
    &&&.image-container {
      display: none;
    }
  `}
`

const FormColumn = styled(Grid.Column)`
  & {
    margin: auto;
  }
`

const FormContainer = styled(Container)`
  ${media.lessThan('large')`
    & {
      padding: 0 1rem;
    }
  `}

  ${media.greaterThan('large')`
    & {
      padding: 0 4.5rem;
    }
  `}

  &&& .field, &&& .button {
    margin-bottom: 2em;
  }

  &&& .field label {
    font-size: 16px;
    font-weight: normal;
  }

  &&& .field label:after {
    display: none;
  }

  &&& img {
    max-width: 65%;
    margin-bottom: 3rem;
  }
`