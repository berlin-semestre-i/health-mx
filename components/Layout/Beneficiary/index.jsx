import React, { Fragment } from 'react'
import SideBar from '../../shared/SideBar/SideBar'
import SearchBar from '../../shared/SearchBar/SearchBar'
import { CustomContainer, MainContainer } from '../../../static/styledComponents'

const Layout = ({ children }) => (
  <Fragment>
    <SideBar className="beneficiary"/>
    <CustomContainer>
      <SearchBar placeholder="Buscar tarea o servicio"/>
      <MainContainer>
        { children }
      </MainContainer>
    </CustomContainer>
  </Fragment>
)

export default Layout