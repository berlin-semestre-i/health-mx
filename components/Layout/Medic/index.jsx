import React from 'react'
import { Container } from 'semantic-ui-react'
import styled from 'styled-components'
import SideBar from '../../shared/SideBar/SideBar'
import SearchBar from '../../shared/SearchBar/SearchBar'

const CustomContainer = styled(Container)`
  &&& {
    margin: 0;
    width: 100%;
  }
`

const MainContainer = styled(Container)`
  &&& {
    margin: 0;
    width: 100%;
    padding: 3rem 3rem 0;
  }
`

const Layout = ({ children }) => (
  <React.Fragment>
    <SideBar background="#3d5170"/>
    <CustomContainer>
      <SearchBar />
      <MainContainer>
        { children }
      </MainContainer>
    </CustomContainer>
  </React.Fragment>
)

export default Layout