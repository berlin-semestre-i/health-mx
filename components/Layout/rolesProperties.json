{
  "medic": {
    "placeholder": "Buscar derechohabiente o tarea",
    "background": "#3d5170"
  },
  "beneficiary": {
    "placeholder": "Buscar tarea o servicio",
    "background": "#0057A3"
  },
  "nurse": {
    "placeholder": "Buscar derechohabiente",
    "background": "#01493D"
  },
  "admin": {
    "placeholder": "Buscar usuario",
    "background": "#561A9B"
  },
  "error": {
    "placeholder": "Error: you have no role",
    "background": "#F65D61"
  }
}