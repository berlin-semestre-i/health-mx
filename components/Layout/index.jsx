import React, { Fragment } from 'react'
import { Container } from 'semantic-ui-react'
import styled from 'styled-components'
import SideBar from '../shared/SideBar/SideBar'
import SearchBar from '../shared/SearchBar/SearchBar'
import { medic, beneficiary, nurse, error, admin } from './rolesProperties.json'

const getPropertiesFromRole = (userRole) => {
  if ( userRole === 'medic' ) {
    return medic
  }
  else if ( userRole === 'beneficiary' ) {
    return beneficiary
  }
  else if ( userRole === 'nurse' ) {
    return nurse
  }
  else if ( userRole === 'admin' ) {
    return admin
  }
  else {
    return error
  }
}

const Layout = ({ children, userRole }) => {
  const layoutProperties = getPropertiesFromRole(userRole)
  return (
    <Fragment>
      <SideBar background={layoutProperties.background}/>
      <GlobalContainer>
        <SearchBar placeholder={layoutProperties.placeholder}/>
        <MainContainer>
          { children }
        </MainContainer>
      </GlobalContainer>
    </Fragment>
  )
}

export default Layout

const GlobalContainer = styled(Container)`
  &&& {
    margin: 0;
    width: 100%;
  }
`

const MainContainer = styled(Container)`
  &&& {
    margin: 0;
    width: 100%;
    padding: 3rem 3rem 0;
  }
`
