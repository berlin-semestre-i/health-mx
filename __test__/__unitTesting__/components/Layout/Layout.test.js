import React from 'react'
import { shallow } from 'enzyme'
import Layout from '../../../../components/Layout'

const defaultRole = 'medic'

describe('Layout', () => {
  it('renders', () => {
    const component = shallow(<Layout userRole={defaultRole}/>)
    expect(component).toMatchSnapshot()
  })
})