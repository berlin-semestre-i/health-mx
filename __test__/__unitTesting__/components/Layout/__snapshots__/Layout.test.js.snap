// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders 1`] = `
<Fragment>
  <withRouter(SideBar)
    userRole="medic"
  />
  <Layout__GlobalContainer>
    <SearchBar
      placeholder="Buscar derechohabiente o tarea"
    />
    <Layout__MainContainer />
  </Layout__GlobalContainer>
</Fragment>
`;
